py/dml/traits.py

```diff
@@ -618,6 +618,9 @@ def __init__(self, site, name, ancestors, methods, params, sessions,
         self.name = name
         self.site = site
 
+        # Set of CompositeObject
+        self.instances = set()
```

Comment:
can we move this to a global mapping `{Trait: {CompositeObject}}` instead? The `Trait` class is currently immutable and I'd prefer to keep it that way.

Reply:
I'm deliberating between
```
class Trait(SubTrait):
   instances = {}
```
and `dml.globals.trait_instances`. Your opinion?

Reply:
Discussed on R.C. `dml.globals` is the way to go.